packages/client/src/fuature/forum/hooks/useNotification.tsx

```diff
@@ -0,0 +1,14 @@
+export async function useNotification (Text:string) {
+
+  if(!("Notification" in window)) {
+    alert("Браузер не поддерживает уведомления!")
+  } else if (Notification.permission === "granted") {
+    new Notification(Text);
+  } else if(Notification.permission !== "denied") {
+    Notification.requestPermission().then((permission) => {
+      if (permission === "granted") {
+        new Notification(Text);
+      }
+    });
+  }
+}
```

Comment:
хуки чуть сложнее обычных функций, их нельзя вызывать везде где хочеться. Например такая конструкция неправильная:
```tsx
<AlertButton type="success" onClick={ () => useNotification("") }>Проверить</AlertButton>
```
потому что ты вызывешь хук внутри onClick, что не правильно (так задумано разработчиками react, и даже будет [ругаться](https://legacy.reactjs.org/warnings/invalid-hook-call-warning.html)).

Поэтому правильнее делать так:
```suggestion
export function useNotification() {
  function notify(text: string) {
    if (!("Notification" in window)) {
      alert("Браузер не поддерживает уведомления!")
    } else if (Notification.permission === "granted") {
      new Notification(Text);
    } else if (Notification.permission !== "denied") {
      Notification.requestPermission().then((permission) => {
        if (permission === "granted") {
          new Notification(text);
        }
      });
    }
  }

  return { notify };
}
```

Место вызовы:
```tsx
const { notify } = useNotification();

<AlertButton type="success" onClick={ () => notify("") }>Проверить</AlertButton>
```